### ISSUE TITLE
Performance issue in 

### ISSUE THREAD

**Issue by ekuleshov** [eobermuhlner/big-math#65]
The `BigDecimalMath.pow()` method is relying on Exception handling to recover from some corner cases.
In my profiling the `BigDecimal.longValueExcact()` can throw `ArithmeticException("Rounding necessary")` which takes a large chunk of the `BigDecimalMath.pow()` execution time.

![image](https://user-images.githubusercontent.com/440954/203866238-90a7fc7f-26d2-45a2-b576-632e6b18164f.png)

The issue seem to be with try/catch code below and I wonder if the exception could be mitigated with some conditions inside try/catch to avoid some common cases of the `ArithmeticException`?

![image](https://user-images.githubusercontent.com/440954/203868573-c73c4dbb-5c7c-453d-b618-45c661c8569d.png)

**Comment by Tostino** [comment]
Yeah I am seeing this show up when attempting to utilize this library in a project of mine.
